lib/galaxy/managers/group_roles.py

```diff
@@ -106,8 +107,13 @@ def _remove_role_from_group(self, trans: ProvidesAppContext, group_role: model.G
 
     def _serialize_group_role(self, encoded_group_id: EncodedDatabaseIdField, role: model.Role):
         encoded_role_id = self._app.security.encode_id(role.id)
-        return {
+        try:
+            url = url_for('role', group_id=encoded_group_id, id=encoded_role_id)
+        except AttributeError:
+            url = "*deprecated attribute not filled in by FastAPI server*"
+
+        return GroupRoleModel(**{
             "id": encoded_role_id,
             "name": role.name,
-            "url": url_for('group_role', group_id=encoded_group_id, id=encoded_role_id)
-        }
+            "url": url
+        })
```

Comment:
There are many ways of initializing a pydantic model, in this case, it is correct as it is, but I will suggest directly setting the attributes like:

```python
return GroupRoleModel(
    id=encoded_role_id,
    name=role.name,
    url=url,
)
```
It is slightly less verbose and the model doesn't need to parse the dictionary :)